layer_metadata_restorer_test.go

```diff
@@ -0,0 +1,322 @@
+package lifecycle_test
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/apex/log"
+	"github.com/apex/log/handlers/discard"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	"github.com/buildpacks/lifecycle"
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/buildpack"
+	"github.com/buildpacks/lifecycle/platform"
+	h "github.com/buildpacks/lifecycle/testhelpers"
+)
+
+func TestLayerMetadataRestorer(t *testing.T) {
+	spec.Run(t, "LayerMetadataRestorer", testLayerMetadataRestorer, spec.Report(report.Terminal{}))
+}
+
+func testLayerMetadataRestorer(t *testing.T, when spec.G, it spec.S) {
+	var (
+		layerDir       string
+		layerAnalyzer  lifecycle.LayerMetadataRestorer
+		layersMetadata platform.LayersMetadata
+		cacheMetadata  platform.CacheMetadata
+		buildpacks     []buildpack.GroupBuildpack
+		skipLayers     bool
+	)
+
+	it.Before(func() {
+		var err error
+
+		layerDir, err = ioutil.TempDir("", "lifecycle-layer-dir")
+		h.AssertNil(t, err)
+		logger := &log.Logger{Handler: &discard.Handler{}}
+		platform := platform.NewPlatform("0.7")
+		layerAnalyzer = lifecycle.NewLayerMetadataRestorer(logger, layerDir, platform, skipLayers)
+	})
+
+	it.After(func() {
+		h.AssertNil(t, os.RemoveAll(layerDir))
+	})
+
+	when("#Analyze", func() {
```

Comment:
I think you wanted it to be `#Restore`, but thinking about these names again, I'm not sure that `Restore` is the best name for the public function in `layer_metsadata_restorer.go`.
Correct me if I'm wrong, but that file is mostly responsible on "analyzing the layers" as described in the last sentence of the [PR](https://github.com/buildpacks/rfcs/blob/main/text/0075-move-analyze-phase.md#how-it-works).
I don't like so much calling that function `AnalyzeLayers` and the file `layer_metadata_analyzer.go`, but maybe you have a better idea.

Reply:
What about `analyzeLayers` -> `restoreLayerMetadata`? It has a nice parity with `restoreStoreTOML`.